optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/reproducer/TestGenCorruptedScoreReproducer.java

```diff
@@ -66,10 +66,10 @@ public boolean isReproducible(TestGenKieSessionJournal journal) {
         } catch (RuntimeException e) {
             if (e.getMessage() != null && e.getMessage().startsWith("No fact handle for ")) {
                 // this is common when removing insert of a fact that is later updated - not interesting
-                logger.debug("    Can't remove insert: {}", e.toString());
+                logger.debug("    Can't remove insert: {}", (Object) e);
```

Comment:
Very nice investigation, thank you for sharing.
The long term solution is indeed that SFL4J PR merged and released. Neither is the case today, so let's go a head with reverting those few lines (only that subset of these changes).